copy.c

```diff
@@ -1,11 +1,55 @@
+#include <fcntl.h>
 #include <stdio.h>
 #include <windows.h>
 
-int wmain(int argc, wchar_t **argv) {
-	DWORD cbSize = (wcslen(argv[1]) + 1) * 2;
+typedef struct node_tag {
+	char buffer[1024];
+	int capacity;
+	struct node_tag *next;
+} node;
+
+int wmain() {
+	_setmode(_fileno(stdin), _O_U8TEXT);
+
+	node *bufferList = malloc(sizeof(node));
+	if (bufferList == NULL) {
+		wprintf(L"Could not allocate memory for reading stdin.\n");
+		return 1;
+	}
+	bufferList->capacity = 0;
+	node *head = bufferList;
+
+	DWORD cbSize = 0;
+	DWORD count;
+	while ((count = fread(head->buffer, 1, sizeof(head->buffer), stdin)) != 0) {
+		cbSize += count;
+		head->capacity = count;
+		head->next = malloc(sizeof(node));
+		if (head->next == NULL) {
+			wprintf(L"Could not allocate memory for reading stdin.\n");
+			return 1;
+		}
+		head->next->capacity = 0;
+	}
+
+	if (ferror(stdin)) {
+		perror("Reading stdin failed");
```

Comment:
`perror` isn't going to work here since `ferror()` and `fread()` don't set `errno` (not sure why they don't, but they don't). If we _really_ wanted to supply a reason for the failure, we'd use `read()` with a file descriptor - however I don't think that's necessary.

Just use `fwprintf(stderr, L"Reading stdin failed")` instead.